### ISSUE TITLE
Add Integration for Tuya Temp-Sensor ZTH02

### ISSUE THREAD

**Issue by tobiraak** [dresden-elektronik/deconz-rest-plugin#6603]
## Device

- Product name: Tuya Zigbee Temp & Humidity Sensor
- Manufacturer: _TZE200_yjjdcqsq
- Model identifier: TS0601
- Device type : 
  - Sensor

## Screenshots

![image](https://user-images.githubusercontent.com/24438079/206597680-5d96163c-3e01-46e5-8d10-d4ce018edb94.png)

![image](https://user-images.githubusercontent.com/24438079/206597833-cd6c9023-3823-4ffc-928d-45f2e669e12c.png)

No other clusters available sadly. 

I am using home assistant, trying to connect the temp sensors via phoscon/deconz but it is not connecting and not showing up in the ui.

**Comment by BabaIsYou** [comment]
A bit surprised that you don't have a Tuya specific cluster (0XEF00) if you don't have Temp (0x402) or Humidity (0x405) standard clusters. Are you sure that joining process for that device was correct ?

After rejoining, and if you have Tuya cluster then you could perhapds try this DDF file 

```
{
  "schema": "devcap1.schema.json",
  "manufacturername": "_TZE200_yjjdcqsq",
  "modelid": "TS0601",
  "product": "ZTH02",
  "sleeper": true,
  "status": "Silver",
  "subdevices": [
    {
      "type": "$TYPE_HUMIDITY_SENSOR",
      "restapi": "/sensors",
      "uuid": [
        "$address.ext",
        "0x01",
        "0x0405"
      ],
      "items": [
        {
          "name": "attr/id"
        },
        {
          "name": "attr/lastannounced"
        },
        {
          "name": "attr/lastseen"
        },
        {
          "name": "attr/manufacturername"
        },
        {
          "name": "attr/modelid"
        },
        {
          "name": "attr/name"
        },
        {
          "name": "attr/swversion"
        },
        {
          "name": "attr/type"
        },
        {
          "name": "attr/uniqueid"
        },
        {
          "name": "config/offset",
          "default": 0
        },
        {
          "name": "config/on"
        },
        {
          "name": "config/reachable"
        },
        {
          "name": "state/humidity",
          "parse": {"fn": "tuya", "dpid": 2, "eval": "Item.val = 10 * Attr.val;" },
          "read": {"fn": "tuya"},
          "default": 0
        },
        {
          "name": "state/lastupdated"
        }
      ]
    },
    {
      "type": "$TYPE_TEMPERATURE_SENSOR",
      "restapi": "/sensors",
      "uuid": [
        "$address.ext",
        "0x01",
        "0x0402"
      ],
      "items": [
        {
          "name": "attr/id"
        },
        {
          "name": "attr/lastannounced"
        },
        {
          "name": "attr/lastseen"
        },
        {
          "name": "attr/manufacturername"
        },
        {
          "name": "attr/modelid"
        },
        {
          "name": "attr/name"
        },
        {
          "name": "attr/swversion"
        },
        {
          "name": "attr/type"
        },
        {
          "name": "attr/uniqueid"
        },
        {
          "name": "config/offset",
          "default": 0
        },
        {
          "name": "config/on"
        },
        {
          "name": "config/reachable"
        },
        {
          "name": "state/lastupdated"
        },
        {
          "name": "state/temperature",
          "parse": {"fn": "tuya", "dpid": 1, "eval": "Item.val = 10 * Attr.val;" },
          "read": {"fn": "none"},
          "default": 0
        }
      ]
    }
  ]
}
```

**Comment by tobiraak** [comment]
My connecting procedure was this: Make the Temp Sensor go into pairing mode -> open up phoscon webapp through Home Assistant and go into pairing mode as well -> since they didn't show up there I went into the deconz app (also via Home Assistant), and here they showed up but without any cluster.

If you have any other procedure that I can try please let me know. I will try the DDF file even though I have no clusters.

**Comment by BabaIsYou** [comment]
Your process seems correct but sometimes devices won’t complete joining process … but can be completed by rejoining it without deleted it first.

Does the DDF of any help (not for connection anyway but for attributes)

Le 10 déc. 2022 à 08:25, Tobias Raak ***@***.***> a écrit :

﻿

My connecting procedure was this: Make the Temp Sensor go into pairing mode -> open up phoscon webapp through Home Assistant and go into pairing mode as well -> since they didn't show up there I went into the deconz app (also via Home Assistant), and here they showed up but without any cluster.

If you have any other procedure that I can try please let me know. I will try the DDF file even though I have no clusters.

**Comment by tobiraak** [comment]
I tried to reconnect again, they won't connect. I deleted the nodes and re-added them but still the same. Also: I am unable to edit the DDF. I have to rightclick the node and "edit DDF" right? No chance to do so unfortunately. Do you have any other idea? I have two of them so I can confirm that they are malfunctioning.

**Comment by BabaIsYou** [comment]
Just to be sure : you deleted it and it won’t pair anymore ?

**Comment by tobiraak** [comment]
They paired again but still the same problem line in the beginning. But I tried your steps first that you provided me in the first place.

**Comment by BabaIsYou** [comment]
> Also: I am unable to edit the DDF. I have to rightclick the node and "edit DDF" right? No chance to do so unfortunately

I'm mot familiar with HA but think you van fond a way to add DDF file with that thread : [https://community.home-assistant.io/t/how-to-install-a-new-deconz-ddf-file/390455](url)

**Comment by nanostrukturtechniker** [comment]
For me the sensor came with a pretty empty battery, maybe this is the reason for the incomplete pairing?

Here are all clusters for me:
![grafik](https://user-images.githubusercontent.com/10961208/209956520-f027f67f-a7a7-42f8-bf9f-7a2a8ea41353.png)

![grafik](https://user-images.githubusercontent.com/10961208/209956702-1afe7085-3b87-46b6-a79b-e78fb74091b2.png)

![grafik](https://user-images.githubusercontent.com/10961208/209956887-eee44437-b2db-437f-ae60-51da6d5ecc8d.png)

![grafik](https://user-images.githubusercontent.com/10961208/209956998-35ea9a58-fc91-4f0e-8987-b418f2de930c.png)

![grafik](https://user-images.githubusercontent.com/10961208/209957184-f3c31b3f-5086-4997-919b-8ccca686fd55.png)

![grafik](https://user-images.githubusercontent.com/10961208/209957358-4681eaed-0ca5-4e5b-a3ec-454c4be3a55b.png)

![grafik](https://user-images.githubusercontent.com/10961208/209957483-20f1dfa3-1423-4188-a381-8bac7cade042.png)

**Comment by Smanar** [comment]
>For me the sensor came with a pretty empty battery, maybe this is the reason for the incomplete pairing?

No it's just the device haven't send value.
To have them press the "read" button, but you need to wake up the device in same time (generaly using short press on the reset button)

**Comment by tobiraak** [comment]
The device is still not able to connect even with all your provided troubleshooting. How ever I tried the Zigbee Plugin that works like a charm. Every Zigbee Device that I have is recognized even the temperature sensors. I just needed to set the DeconZ Bridge in the Zigbee Plugin Settings.

I don't know why the DeconZ/Phoscon Plugin is not working but for everyone with the same issues I can recommend switching from the DeconZ Plugin to the Zigbee Plugin if you are using Home Assistant. 
I will close this issue since I got it working with a different approach.
